[Typography foundations] Deprecate existing typography components on polaris.shopify.com

laurkim opened the pull request:
~### ⛔️ DO NOT MERGE~
~Alpha release of `Text` component needs to [ship](https://github.com/Shopify/polaris/pull/6675) first.~

### WHY are these changes introduced?

Resolves #6513.

Once the alpha Text component is released, the existing typography component (`DisplayText`, `Heading`, `Subheading`, `Caption`, `TextStyle`, and `VisuallyHidden`) docs need to be updated.

### WHAT is this pull request doing?

- Deprecates `DisplayText`, `Heading`, `Subheading`, `Caption`, `TextStyle`, and `VisuallyHidden` pages
- Adds mapping from ^ components to `Text` component
- Removes examples for all the above components
- Updates `[components].tsx` to only render `ComponentExamples` if they exist
- Updates `[components].tsx` to only render `PropTables` if the component isn't deprecated
    <details>
      <summary>Heading page — before</summary>
      <img src="https://user-images.githubusercontent.com/26749317/183943418-47eae187-95d8-41a2-92ba-fda390028f58.png" alt="Heading page — before">
    </details>
    <details>
      <summary>Heading page — after</summary>
      <img src="https://user-images.githubusercontent.com/26749317/183943410-f2c38c22-9776-4964-95a0-645299216639.png" alt="Heading page — after">
    </details>

<!-- ℹ️ Delete the following for small / trivial changes -->

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

laurkim committed:
[polaris.shopify.com] Update component to hide examples and PropsTable for deprecated components

laurkim committed:
[Typography foundations] Deprecate `Heading` component on styleguide

laurkim committed:
[Typography foundations] Deprecate `DisplayText` component on styleguide

laurkim committed:
[Typography foundations] Deprecate `Subheading` component on styleguide

laurkim committed:
[Typography foundations] Deprecate `Caption` component on styleguide

laurkim committed:
[Typography foundations] Deprecate `TextStyle` component on styleguide

laurkim committed:
[Typography foundations] Deprecate `VisuallyHidden` component on styleguide

laurkim committed:
Add changeset

aveline commented on polaris.shopify.com/content/components/caption/index.md:
```diff
@@ -15,10 +15,18 @@ keywords:
-    description: Use to provide details in situations where content is compact and space is tight.
+status:
+  value: Deprecated
+  message: This component is no longer supported. Please use the alpha Text component instead.
```
Should we just refer to the `Text` component rather than `alpha Text` component?

laurkim replied:
I was thinking of leaving it as `alpha Text` until we remove the `alpha` flag from the Text component page. I don't have a super strong opinion on this though, what do you think @aveline?

alex-page replied:
I agree with @aveline because we are starting to remove these old components. Users should be confident in using the new `Text` component.

alex-page replied:
We can keep it as `alpha` tagged in the sidebar though.

aveline replied:
Yes I think referring to `Text` in the docs but leaving the `alpha` tag in the sidebar makes sense.

laurkim replied:
Sounds good, will update thanks for feedback! 👍

laurkim committed:
[Typography foundations] Update messaging re: `alpha Text component`

laurkim committed:
[Typography foundations] Update legacy mapping to Text component

Based off the work to modify tshirt sizes.
https://github.com/Shopify/polaris/issues/6967
